library/alloc/src/vec.rs

```diff
@@ -2730,6 +2730,14 @@ impl<T> Iterator for IntoIter<T> {
     fn count(self) -> usize {
         self.len()
     }
+
+    #[inline]
+    fn last(self) -> Option<Self::Item>
+    where
+        Self: Sized,
+    {
+        if self.ptr == self.end { None } else { unsafe { Some(ptr::read(self.end.offset(-1))) } }
```

Comment:
This doesn't seemed to handle zero sized type, `self.end.offset(-1)` will probably read the wrong address.